spec/ruby/core/numeric/helpers/step.rb

```diff
@@ -0,0 +1,13 @@
+class Object
+  def get_args(args_type, *args)
```

Comment:
Besides the step = 0 vs by: 0 difference and the ArgumentError raising when repeating arguments when using positional and keyword arguments, the results should always be the same.

The idea behind this was to avoid duplicating the specs 3 times where the only thing that changes is how you pass the arguments.

I can thing on other options, and I'm open to suggestions, being it a different approach or clarifying this method to avoid confusion, but I think that duplicating the tests is the worst one.